### DISCUSSION TITLE
A thought for 2.0 - Rotary Encoder Support

### DISCUSSION THREAD

**Discussion by Hidyman** [tjko/fanpico#23] [Ideas]
For some background:
I'm running TrueNAS Scale, and as of right now there is no way (or easy way) to use the HDD temps to control the chassis fan speeds (This MB has 6 chassis fan headers basically going to waste). Looking for a solution brought me to this project. 
My use case is inside a server case, so it's basically in a faraday cage making the W version useless.

I think it would be amazing to add support for a simple 5 pin rotary encoder (the kind that has a switch when you push it).
This could be used to adjust/add points on the fan curves, or basically change any configuration setting, and perhaps change between several display modes. I could 3D print a small case to hold the display and the rotary encoder allowing me to put it on the front of the server to allow monitoring and control.

As to the limited amount of inputs and outputs:
I believe a rotary encoder would need 2 inputs for the encoder and one for the switch, so that would require losing 2 MB inputs or 2 Fan outputs or a combination of 1 of each. 
My server case has 3 drive bays with a fan in front of and behind each bay. So, in my case, it would be ideal to lose one input and one output, and use the extra line for a third temperature sensor so that I can monitor each bay individually, while still keeping the inputs in the case that TrueNAS starts supporting fan speed control via the HDD temperature reporting. 

For the ultimate in flexibility, maybe there could be a series of headers that could be bridged one way or another to change a couple of the inputs/outputs to allow the use of a rotary encoder and/or adding a third temperature sensor.

**Comment by tjko** [comment]
In your case you should be able to use the USB (serial) connection and have small script/program polling hard drive temperatures and feeding this to fanpico? 

Maybe fanpico firmware could support 'external' temperature registers that external program could periodically update, and fans could be configured to use those as source, or something... (with current firmware external program could also directly adjust the fan PWM value, if wan fas set to 'fixed' speed).

Latest PCB has I2C exposed, so it would allow adding I2C based rotary encoder(s) easily, along with other I2C sensor(s) (like temperature sensors):
https://learn.adafruit.com/adafruit-i2c-qt-rotary-encoder?view=all

**Comment by tjko** [comment]
Adding another 74HC4051 multiplexer in "parallel" (sharing the address pins) with the existing one, should allow 8 temperature sensors, while also freeing one pin on the Pico.  Not sure if anyone would ever need 8 temperature inputs, but this would be easy to implement and really would not add significant cost either.
